File: examples/csharp/factory-method/CalculationUnit.Factory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FactoryMethod
{
    internal class CalculationUnitFactory
    {
        public enum CalculationUnitType
        {
            Totalizer,
            Mediator
        }
        public static ICalculationUnit CreateCalculationUnit(CalculationUnitType cuType)
        {
            switch(cuType)
            {
                case CalculationUnitType.Totalizer:
                    {
                        return new CalculationUnitTotalizer();                        
                    }
                case CalculationUnitType.Mediator:
                    {
                        return new CalculationUnitMediator();
                    }
                default:
                    {
                        throw new ArgumentException("Calculation unit type not implemented.");
                    }
            }
        }
    }
}

File: examples/python/factory-method/src/calculation_unit_factory.py
from enum import Enum
from calculation_unit_totalizer import CalculationUnitTotalizer
from calculation_unit_mediator import CalculationUnitMediator

class CalculationUnitTypes(Enum):
    totalizer = 0
    mediator = 1

def create_calculation_unit(calculation_unit_type):
    if calculation_unit_type == CalculationUnitTypes.totalizer:
        return CalculationUnitTotalizer()
    elif calculation_unit_type == CalculationUnitTypes.mediator:
        return CalculationUnitMediator()
    else:
        raise Exception('Calculation unit type not implemented.')